lib/compiler/swc/swc-compiler.ts

```diff
@@ -49,29 +49,23 @@ export class SwcCompiler extends BaseCompiler {
       appName,
     );
 
-    if (extras.watch) {
-      if (extras.typeCheck) {
-        this.runTypeChecker(configuration, tsConfigPath, appName, extras);
-      }
-      await this.runSwc(swcOptions, extras, swcrcFilePath);
+    if (extras.typeCheck) {
+      this.runTypeChecker(configuration, tsConfigPath, appName, extras);
```

Comment:
This introduces a breaking change - in watch mode, `runTypeChecker` shouldn't be awaited but when it's turned off, we have to use `await`

Reply:
oh I didn't think it was expected to run swc without waiting for typecheck, which is kind of the expected behavior with typeCheck. I'll re-commit with just the fix.